src/js/models/tools/superselect/vector.js

```diff
@@ -107,6 +109,8 @@ define(function (require, exports, module) {
             flux.actions.tools.select(toolStore.getToolByID("newSelect"));
         }
     };
+    
+    SuperSelectVectorTool.prototype.toolOverlays = [SuperselectOverlay, SuperselectVectorMaskOverlay];
```

Comment:
So when user is in direct edit mode of a vector (no masks), they're going to see bounds of other layers highlight? If there is a flicker, there must be a bug somewhere in the code unrelated to this.